File: app/src/main/java/ru/dmitrybochkov/rates/di/NetworkServiceModule.java
package ru.dmitrybochkov.rates.di;

import javax.inject.Singleton;

import androidx.annotation.NonNull;
import dagger.Module;
import dagger.Provides;
import ru.dmitrybochkov.rates.data.api.v1.NetworkService;
import ru.dmitrybochkov.rates.data.api.RetrofitGenerator;

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */
@Module
public class NetworkServiceModule {

    @Provides
    @NonNull
    @Singleton
    NetworkService provideNetworkService() {
        return RetrofitGenerator.createService();
    }

}

File: app/src/main/java/ru/dmitrybochkov/rates/di/NetworkServiceModule.kt
package ru.dmitrybochkov.rates.di

import dagger.Module
import dagger.Provides
import ru.dmitrybochkov.rates.data.api.RetrofitGenerator
import ru.dmitrybochkov.rates.data.api.v1.NetworkService
import javax.inject.Singleton

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */
@Module
class NetworkServiceModule {

    @Provides
    @Singleton
    internal fun provideNetworkService(): NetworkService {
        return RetrofitGenerator.createService()
    }

}
